TROUBLESHOOT.md

```diff
@@ -0,0 +1,165 @@
+
+## Overview
+When problems occur, debug tools enable users to quickly pinpoint the root cause, whether it's 
+within the connector itself, the BigQuery API, or the network infrastructure.
+Detailed logs, traces, and debugging capabilities help you effectively troubleshoot issues, 
+reducing the time and effort required to resolve them.
+This document is a debugging manual highlighting possible issues that could be faced by the users 
+and how to troubleshoot them. It also aids in identifying the error causes and proposed steps for 
+mitigating them.
+
+<i>Note: This is a running document containing the issues reported to the Developers of the 
+[Flink - BigQuery Connector](https://github.com/GoogleCloudDataproc/flink-bigquery-connector). 
+Please feel free to contribute as per any new issues that may arise.</i>
+
+## Flink Metrics
+### Writing to BigQuery
+As an effort to increase observability, the Flink-BigQuery Connector Team provides support to 
+collect and report Flink Metrics for a Flink Application.
+The details of the metrics supported so far are available in the 
+[README](https://github.com/GoogleCloudDataproc/flink-bigquery-connector/blob/main/README.md#flink-metrics).
+
+An overview of records seen by the writer, number of records successfully written to 
+BigQuery would enable users to track the flow of records through their application. 
+Comparison of the counts would help troubleshoot if records are being seen by the Sink (Writer) 
+at all, are being serialized and send to the Write API and if BigQuery is able to write 
+these records.
+
+See Flink’s documentation on 
+[Metric Reporters](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/metric_reporters/) 
+to deploy the reporter most conveniently as per user’s need.
+
+## General Debugging
+### Records are not being written to BigQuery
+With the help of metrics available as a part of 0.4.0 release of the connector. 
+Users should be able to track the number of records that enter the sink(writer) and the 
+number of records successfully written to BigQuery. 
+If records are not being written to BigQuery, then records are stuck in either of the two phases:
+#### The records are not arriving at the sink
+- The problem lies with the pipeline, the previous chain of subtasks that are performed before 
+sink is called.
+- The pipeline is not processing the passing the records forward for the sink to serialize and 
+write to BigQuery.
+#### The records are arriving at the sink but not being successfully written to BigQuery.
+- Check the logs or error message for the following errors:
+     - #### `BigQuerySerializationException`
+     - This message illustrates that the record(s) could not be serialized by the connector. 
+     - The error message would also contain the actual cause for the same.
+     - #### `BigQueryConnectorException`
+     - Refer to Java Docs for the BigQuery Storage Write API proper usage and the errors that might
+arise from the same.
```

Comment:
This is not good enough for readers. Lets look at why the connector can throw `BigQueryConnectorException` and highlight those potential causes (with fixes if possible)